intelmq/bots/experts/splunk_saved_search/expert.py

```diff
@@ -0,0 +1,192 @@
+
+        self.set_request_parameters()
+
+        self.http_header.update({"Authorization": f"Bearer {self.auth_token}"})
+
+        self.session = utils.create_request_session(self)
+        self.session.keep_alive = False
+
+    def update_event(self, event, search_result):
+        self.logger.info("Updating event: %s",
+                         dict([(field, search_result[field]) for field in self.result_fields]))
+        for result, field in self.result_fields.items():
+            event.add(field, search_result[result], overwrite=self.overwrite)
+
+    def process(self):
+        event = self.receive_message()
+
+        self.logger.info("Received event, searching for %s",
+                         dict([(parameter, event[field]) for field, parameter in self.search_parameters.items()]))
+
+        query = f'|savedsearch "{self.saved_search}"'
+        for field, parameter in self.search_parameters.items():
+            query += f' "{parameter}"="{event[field]}"'
+
+        self.logger.debug("Query: %s", query)
+
+        req = self.session.post(url=self.url + "/services/search/jobs",
+                                data={"output_mode": "json", "search": query},
+                                timeout=self.http_timeout_sec)
+
+        if not req.ok:
+            self.logger.error("Error starting search job: %r",
+                              req.text)
+            req.raise_for_status()
+
+        jobid = req.json()['sid']
+
+        self.logger.debug("Started search, job id: %i", jobid)
+
+        # Even the simplest search is never ready immediately, so to
+        # avoid polling and waiting a full retry_interval for every
+        # search, sleep briefly here in the hope that a simple search
+        # will be ready afterwards.
+        time.sleep(1)
+
+        results_ready = False
+        while not results_ready:
+            req = self.session.get(url=self.url +
+                                   "/services/search/jobs/" + jobid + "/results/",
+                                   data={"output_mode": "json"},
+                                   timeout=self.http_timeout_sec)
+
+            if req.status_code == 200:
+                results_ready = True
+            elif req.status_code == 204:
+                results_ready = False
+                self.logger.debug(f"Results not ready, sleeping for {self.retry_interval} seconds before retrying")
+                time.sleep(self.retry_interval)
+            else:
+                self.logger.error("Error getting search results: %s", req.text)
+                req.raise_for_status()
+
+        hits = req.json()['results']
+        if len(hits) == 0:
+            if "warn" in self.not_found:
+                self.logger.warning("No results returned")
+            if "drop" in self.not_found:
+                self.logger.debug("Dropping message")
+                self.acknowledge_message()
+            if "send" in self.not_found:
```

Comment:
This can be an `elif`

Reply:
It can, however I find the code easier to read this way. If you strongly prefer `elif`, I'll change it.